File: src/NetFxCSharp/Facesso.Interfaces/Kannegiesser SQL/KannegiesserSQLProductionDataImportClasses.vb
Imports System.Data.OleDb
Imports System.Data.SqlClient

<FacessoImportFilterName("Kannegiesser SQL Produktionsdatenimport", FacessoImportType.WorkGroupData, FacessoInterfaceBrand.KannegiesserSQLProductionData)> _
Public Class KannegiesserSQLProductionDataImportTaskElement
    Inherits FacessoProductionDataImportTaskItemBase

    Private myKannegiesserSQLConnectionString As String
    Private myCurrDate As Date
    Private myCurrOrgData As DataTable
    Private myCurrFacData As ProductionDataTable

    Public Property KannegiesserDeviceID As String

    Public Overrides Function ConfigureImportFilter() As System.Windows.Forms.DialogResult
        Dim locFrm As New frmKannegiesserSQLProdDataConfigDialog
        Return locFrm.HandleDialog(Me)
    End Function

    Public Overrides Function GetData(ByVal ProductionDate As Date, ByVal Shift As ShiftCombination) As IImportResultTable
        retrieveDataForDate(ProductionDate)
        Return myCurrFacData
    End Function

    Public Overrides ReadOnly Property ImportType() As FacessoImportType
        Get
            Return FacessoImportType.WorkGroupData
        End Get
    End Property

    Public Overrides ReadOnly Property InterfaceBrand() As FacessoInterfaceBrand
        Get
            Return FacessoInterfaceBrand.KannegiesserSQLProductionData
        End Get
    End Property

    Public Overrides Function ConfigureGenericInterface() As System.Windows.Forms.DialogResult
        Return Windows.Forms.DialogResult.OK
    End Function

    Public Overrides ReadOnly Property IsGenericInterfaceConfigured() As Boolean
        Get
            Return True
        End Get
    End Property

    Public Overrides Function AssembleConversionItems() As FacessoConversionItemsBase
        If KannegiesserSQLConnectionString Is Nothing Then
            Return Nothing
        End If

        Dim locConversionItems As FacessoConversionItemsBase
        locConversionItems = New FacessoConversionItemsBase


        Dim oc = New KannegiesserDataContext()
        Dim artpar = From prgItems In oc.GetArticles

        For Each artItems In artpar
            locConversionItems.Add(New FacessoConversionItemBase(artItems.ArticleID, artItems.ArticleName))
        Next
        Return locConversionItems
    End Function

    'Dass es hier zwei Eigenschaften gibt, die den SQL-Connection-String zurückliefern, hat historische Gründe.
    Public ReadOnly Property ConnectionString() As String
        Get
            Return myKannegiesserSQLConnectionString
        End Get
    End Property

    Public Property KannegiesserSQLConnectionString() As String
        Get
            Return myKannegiesserSQLConnectionString
        End Get
        Set(ByVal value As String)
            myKannegiesserSQLConnectionString = value
        End Set
    End Property

End Class

File: src/NetFxCSharp/Facesso.Interfaces/Kannegiesser SQL/KannegiesserSQLProductionDataImportClasses.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Data.OleDb;
using System.Data.SqlClient;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace Facesso.Interfaces
{
    [FacessoImportFilterName("Kannegiesser SQL Produktionsdatenimport", FacessoImportType.WorkGroupData, FacessoInterfaceBrand.KannegiesserSQLProductionData)]
    public partial class KannegiesserSQLProductionDataImportTaskElement : FacessoProductionDataImportTaskItemBase
    {
        private string myKannegiesserSQLConnectionString;
        private System.DateTime myCurrDate;
        private DataTable myCurrOrgData;
        private ProductionDataTable myCurrFacData;
        public string KannegiesserDeviceID { get; set; }

        public override System.Windows.Forms.DialogResult ConfigureImportFilter()
        {
            frmKannegiesserSQLProdDataConfigDialog locFrm = new frmKannegiesserSQLProdDataConfigDialog();
            return locFrm.HandleDialog(this);
        }

        public override IImportResultTable GetData(System.DateTime ProductionDate, ShiftCombination Shift)
        {
            retrieveDataForDate(ProductionDate);
            return myCurrFacData;
        }

        public override FacessoImportType ImportType
        {
            get
            {
                return FacessoImportType.WorkGroupData;
            }
        }

        public override FacessoInterfaceBrand InterfaceBrand
        {
            get
            {
                return FacessoInterfaceBrand.KannegiesserSQLProductionData;
            }
        }

        public override System.Windows.Forms.DialogResult ConfigureGenericInterface()
        {
            return System.Windows.Forms.DialogResult.OK;
        }

        public override bool IsGenericInterfaceConfigured
        {
            get
            {
                return true;
            }
        }

        public override FacessoConversionItemsBase AssembleConversionItems()
        {
            if (KannegiesserSQLConnectionString == null)
            {
                return null;
            }

            FacessoConversionItemsBase locConversionItems = default(FacessoConversionItemsBase);
            locConversionItems = new FacessoConversionItemsBase();
            var oc = new KannegiesserDataContext();
            var artpar = (
                from prgItems in oc.GetArticles()select prgItems);
            foreach (var artItems in artpar)
            {
                locConversionItems.Add(new FacessoConversionItemBase(artItems.ArticleID, artItems.ArticleName));
            }

            return locConversionItems;
        }

        //Dass es hier zwei Eigenschaften gibt, die den SQL-Connection-String zurückliefern, hat historische Gründe.
        public string ConnectionString
        {
            get
            {
                return myKannegiesserSQLConnectionString;
            }
        }

        public string KannegiesserSQLConnectionString
        {
            get
            {
                return myKannegiesserSQLConnectionString;
            }

            set
            {
                myKannegiesserSQLConnectionString = value;
            }
        }
    }
}